pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -511,69 +513,73 @@ func (b *BridgeBindMechanism) startDHCP() error {
 	return nil
 }
 
-func (b *BridgeBindMechanism) preparePodNetworkInterface() (api.Interface, error) {
+func (b *BridgeBindMechanism) preparePodNetworkInterface() error {
 	// Set interface link to down to change its MAC address
 	if err := b.handler.LinkSetDown(b.podNicLink); err != nil {
 		log.Log.Reason(err).Errorf("failed to bring link down for interface: %s", b.podInterfaceName)
-		return api.Interface{}, err
+		return err
 	}
 
-	tapDeviceName := generateTapDeviceName(b.podInterfaceName)
+	b.tapDeviceName = generateTapDeviceName(b.podInterfaceName)
```

Comment:
I think we can do this on creation in `getPhase2Binding` (therefore, not mutate it later on).

Do you think it will be more complete to do so for all bindings, so we will not have the tap device name generated differently for each binding type? In practice, we can calculate it once, and populate the attribute on creation for each binding that uses a tap device. This way it will be clear that the naming is the same for all bindings.
WDYT?

Reply:
> I think we can do this on creation in `getPhase2Binding` (therefore, not mutate it later on).

I'd usually agree with you, but I prefer to initialize stuff when I need it, not 3 methods away from it. But if you insist, I can do it.

> 
> Do you think it will be more complete to do so for all bindings, so we will not have the tap device name generated differently for each binding type? In practice, we can calculate it once, and populate the attribute on creation for each binding that uses a tap device. This way it will be clear that the naming is the same for all bindings.
> WDYT?

I disagree; I'd rather not add code & attributes to all bind mechanisms when only a single one requires it because one function that generates names for a device is not idempotent.

I don't see a reason for making a pattern out of this.

Reply:
> I'd usually agree with you, but I prefer to initialize stuff when I need it, not 3 methods away from it. But if you insist, I can do it.

IMO if the info exists on construction, it should be done there. It is a state, and there is no apparent reason to fill it later. But it does not merit insisting from my part, especially when you do not agree to the second part.

> 
> I disagree; I'd rather not add code & attributes to all bind mechanisms (well, all that use tap devices) when only a single one requires it because one function that generates names for a device is not idempotent.
> 
> I don't see a reason for making a pattern out of this.

It is not about a pattern, but about structuring the code such that commonality is performed in one way. In this case, the commonality is not full (not all bindings have a tap), so I will not argue further here. But I would like to see a code structure that supports generating the domain spec which is partially common and partially specific to the type.
Anyway, this is futuristic and we can leave it for another refactoring day.